### PR TITLE
Bluetooth:tester: Support BREDR GAP, SDP and L2CAP auto-pts test

### PR DESCRIPTION
**Pull request by CherQin**
Update bluetooth bt_tester to support bredr GAP, SDP and L2CAP auto-pts test.

To support these BREDR auto-pts test, auto-pts repo should also be updated, here's my for auto-pts repo update: https://github.com/auto-pts/auto-pts/pull/1131

### CODE REVIEW

**Comment by CherQin** [comment]
To support these BREDR auto-pts test, auto-pts repo should also be updated, here's my for auto-pts repo update: https://github.com/auto-pts/auto-pts/pull/1131

File: tests/bluetooth/tester/src/btp/btp_gap.h [hunk 1]
```diff
@@ -141,9 +142,12 @@ struct btp_gap_start_discovery_cmd {
 #define BTP_GAP_STOP_DISCOVERY			0x0d
 
 #define BTP_GAP_CONNECT				0x0e
+#define BTP_GAP_CONNECT_LE			0x00
+#define BTP_GAP_CONNECT_BREDR		0x01
 struct btp_gap_connect_cmd {
 	bt_addr_le_t address;
 	uint8_t own_addr_type;
+    uint8_t transport;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp/btp_gap.h:150]: Formatting issue here: the indentation is different (spaces vs tab) from the rest of the struct

File: tests/bluetooth/tester/src/btp/btp_gap.h [hunk 2]
```diff
@@ -319,6 +323,11 @@ struct btp_gap_padv_sync_transfer_recv_cmd {
 	uint8_t flags;
 } __packed;
 
+#define BTP_SM_AUTH				0x2a
+struct btp_sm_auth_cmd {
+	bt_addr_le_t address;
+    uint8_t   useh7;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp/btp_gap.h:329]: Same here. First line uses a tab and the second uses spaces.

File: include/zephyr/bluetooth/hci_types.h [hunk 3]
```diff
@@ -630,11 +630,26 @@ struct bt_hci_cp_host_num_completed_packets {
...
+
 #define BT_HCI_OP_WRITE_INQUIRY_MODE            BT_OP(BT_OGF_BASEBAND, 0x0045) /* 0x0c45 */
 struct bt_hci_cp_write_inquiry_mode {
 	uint8_t  mode;
 } __packed;
 
+#define BT_HCI_OP_WRITE_EXTENDED_INQUIRY_RESPONSE   BT_OP(BT_OGF_BASEBAND, 0x0052) /* 0x0c52 */
+struct bt_hci_cp_write_extended_inquiry_response {
+	uint8_t fec_required;
+	uint8_t Extended_Inquiry_Response[240];
```
Comment (jhedberg) [inline include/zephyr/bluetooth/hci_types.h:650]: This doesn't follow the Zephyr coding style with the capitalization. Why not just `uint8_t eir[240];`?

File: tests/bluetooth/tester/src/btp_gap.c [hunk 4]
```diff
@@ -106,7 +118,27 @@ static void le_connected(struct bt_conn *conn, uint8_t err)
...
+	if (BT_CONN_TYPE_BR == info.type)
+	{
+		bt_conn_set_security(conn, BT_SECURITY_L2);
+	}
+#endif /* CONFIG_BT_CLASSIC */
+#endif
+
+#if defined(CONFIG_BT_CLASSIC)
+	if (BT_CONN_TYPE_BR == info.type)
+	{
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:132]: The `{` should go on the same line as the `if`

File: tests/bluetooth/tester/src/btp_gap.c [hunk 5]
```diff
@@ -106,7 +118,27 @@ static void le_connected(struct bt_conn *conn, uint8_t err)
...
+		bt_br_set_discoverable(false);
+		bt_br_set_connectable(false);
+	}
+#endif /* CONFIG_BT_CLASSIC */
+
+	if (info.le.dst != NULL)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:138]: Same here

File: tests/bluetooth/tester/src/btp_gap.c [hunk 6]
```diff
@@ -144,7 +176,7 @@ static void le_identity_resolved(struct bt_conn *conn, const bt_addr_le_t *rpa,
 }
 
 static void le_param_updated(struct bt_conn *conn, uint16_t interval,
-			     uint16_t latency, uint16_t timeout)
+				 uint16_t latency, uint16_t timeout)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:179]: What's this change about?

File: tests/bluetooth/tester/src/btp_gap.c [hunk 7]
```diff
@@ -170,6 +202,28 @@ static bool le_param_req(struct bt_conn *conn, struct bt_le_conn_param *param)
...
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+
+	if (NULL == conn)
+	{
+		conn = bt_conn_lookup_addr_br(&cp->address.a);
+	}
+
+#if 0
+	bt_smp_set_ct2(conn, cp->useh7 ? 1:0);
+#endif
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:221]: Please use a proper Kconfig symbol, or just remove these three lines.

File: tests/bluetooth/tester/src/btp_gap.c [hunk 8]
```diff
@@ -185,7 +239,7 @@ static void le_security_changed(struct bt_conn *conn, bt_security_t level,
 		sec_ev.sec_level = level;
 
 		tester_event(BTP_SERVICE_ID_GAP, BTP_GAP_EV_SEC_LEVEL_CHANGED,
-			     &sec_ev, sizeof(sec_ev));
+				 &sec_ev, sizeof(sec_ev));
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:242]: What's this change about?

File: tests/bluetooth/tester/src/btp_gap.c [hunk 9]
```diff
@@ -194,13 +248,13 @@ static void le_security_changed(struct bt_conn *conn, bt_security_t level,
 		 * This means bond is lost and we restart pairing to re-bond
 		 */
 		if (bt_conn_get_info(conn, &info) == 0 &&
-		    info.role == BT_CONN_ROLE_CENTRAL) {
+			info.role == BT_CONN_ROLE_CENTRAL) {
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:251]: This also seems unrelated and wrong. Split lines should be aligned right after the opening parenthesis of the previous line, using 0-N tabs + 0-7 spaces. Looks to me like the line was already correct before this change.

File: tests/bluetooth/tester/src/btp_gap.c [hunk 10]
```diff
@@ -194,13 +248,13 @@ static void le_security_changed(struct bt_conn *conn, bt_security_t level,
...
 			LOG_DBG("Bond lost");
 
 			bt_addr_le_copy(&bond_ev.address, addr);
 
 			tester_event(BTP_SERVICE_ID_GAP, BTP_GAP_EV_BOND_LOST,
-				     &bond_ev, sizeof(bond_ev));
+					 &bond_ev, sizeof(bond_ev));
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:257]: Same here. Even if there was a need to fix coding style issues such changes should not be mixed into commits which contain functional changes.

File: tests/bluetooth/tester/src/btp_gap.c [hunk 11]
```diff
@@ -267,7 +321,7 @@ static uint8_t supported_commands(const void *cmd, uint16_t cmd_len,
 }
 
 static uint8_t controller_index_list(const void *cmd, uint16_t cmd_len,
-				     void *rsp, uint16_t *rsp_len)
+					 void *rsp, uint16_t *rsp_len)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:324]: Same thing here. Just remove all these indentation changes, since they're unrelated and wrong.

File: tests/bluetooth/tester/src/btp_gap.c [hunk 12]
```diff
@@ -280,7 +334,7 @@ static uint8_t controller_index_list(const void *cmd, uint16_t cmd_len,
 }
 
 static uint8_t controller_info(const void *cmd, uint16_t cmd_len,
-			       void *rsp, uint16_t *rsp_len)
+				   void *rsp, uint16_t *rsp_len)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:337]: Here too

File: tests/bluetooth/tester/src/btp_gap.c [hunk 13]
```diff
@@ -342,7 +396,7 @@ static const char *oob_config_str(int oob_config)
 #endif /* !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY) */
 
 static void oob_data_request(struct bt_conn *conn,
-			     struct bt_conn_oob_info *oob_info)
+				 struct bt_conn_oob_info *oob_info)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:399]: Here too

File: tests/bluetooth/tester/src/btp_gap.c [hunk 14]
```diff
@@ -365,13 +419,13 @@ static void oob_data_request(struct bt_conn *conn,
 
 		struct bt_le_oob_sc_data *oobd_local =
 			oob_info->lesc.oob_config != BT_CONN_OOB_REMOTE_ONLY ?
-				      &oob_sc_local.le_sc_data :
-				      NULL;
+					  &oob_sc_local.le_sc_data :
+					  NULL;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:423]: Here too

File: tests/bluetooth/tester/src/btp_gap.c [hunk 15]
```diff
@@ -365,13 +419,13 @@ static void oob_data_request(struct bt_conn *conn,
...
 
 		struct bt_le_oob_sc_data *oobd_remote =
 			oob_info->lesc.oob_config != BT_CONN_OOB_LOCAL_ONLY ?
-				      &oob_sc_remote.le_sc_data :
-				      NULL;
+					  &oob_sc_remote.le_sc_data :
+					  NULL;
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:428]: And here

File: tests/bluetooth/tester/src/btp_gap.c [hunk 16]
```diff
@@ -381,7 +435,7 @@ static void oob_data_request(struct bt_conn *conn,
 		}
 
 		if (oobd_local &&
-		    !bt_addr_le_eq(info.le.local, &oob_sc_local.addr)) {
+			!bt_addr_le_eq(info.le.local, &oob_sc_local.addr)) {
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:438]: And here

File: tests/bluetooth/tester/src/btp_gap.c [hunk 17]
```diff
@@ -417,7 +471,7 @@ static void oob_data_request(struct bt_conn *conn,
 
 #if !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY)
 static uint8_t get_oob_sc_local_data(const void *cmd, uint16_t cmd_len,
-				     void *rsp, uint16_t *rsp_len)
+					 void *rsp, uint16_t *rsp_len)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:474]: And here

File: tests/bluetooth/tester/src/btp_gap.c [hunk 18]
```diff
@@ -431,7 +485,7 @@ static uint8_t get_oob_sc_local_data(const void *cmd, uint16_t cmd_len,
 }
 
 static uint8_t set_oob_sc_remote_data(const void *cmd, uint16_t cmd_len,
-				      void *rsp, uint16_t *rsp_len)
+					  void *rsp, uint16_t *rsp_len)
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:488]: And here. There seems to be so many places that I won't comment on all of them, but please fix them everywhere.

File: tests/bluetooth/tester/src/btp_gap.c [hunk 19]
```diff
@@ -547,8 +612,34 @@ int tester_gap_stop_ext_adv(void)
...
+struct k_work_delayable limited_discoverable_work;
+
+void limited_discoverable_work_handler(struct k_work *work)
+{
+	struct btp_gap_set_discoverable_rp rsp;
+
+	int err = bt_br_set_discoverable(false);
+
+	if (err == 0)
+	{
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:625]: `if (err == 0) {`

File: tests/bluetooth/tester/src/btp_gap.c [hunk 20]
```diff
@@ -547,8 +612,34 @@ int tester_gap_stop_ext_adv(void)
...
+		err = bt_br_set_connectable(false);
+	}
+
+	if (err == 0)
+	{
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:630]: `if (err == 0) {`

File: tests/bluetooth/tester/src/btp_gap.c [hunk 21]
```diff
@@ -572,14 +663,135 @@ static uint8_t set_discoverable(const void *cmd, uint16_t cmd_len,
...
+
+		struct net_buf *buf = NULL;
+		struct bt_hci_cp_write_class_of_device *cod;
+		struct bt_hci_cp_write_current_iac_lap *iac;
+		struct bt_hci_cp_write_extended_inquiry_response *eir_response;
+		uint32_t cod_value = 0x000000U;
+		uint32_t num_iacs = 1U;
+
+		if (BTP_GAP_LIMITED_DISCOVERABLE == cp->discoverable)
+		{
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:679]: Please fix the style of all of your if-statments

File: tests/bluetooth/tester/src/btp_gap.c [hunk 22]
```diff
@@ -572,14 +663,135 @@ static uint8_t set_discoverable(const void *cmd, uint16_t cmd_len,
...
+
+		buf = bt_hci_cmd_create(BT_HCI_OP_WRITE_CLASS_OF_DEVICE, sizeof(*cod));
+		if (buf != NULL)
+		{
+			cod = net_buf_add(buf, sizeof(*cod));
+			sys_put_le24(cod_value, &cod->class_of_device[0]);
+			err = bt_hci_cmd_send_sync(BT_HCI_OP_WRITE_CLASS_OF_DEVICE, buf, NULL);
+		}
+		else
+		{
```
Comment (jhedberg) [inline tests/bluetooth/tester/src/btp_gap.c:692]: `} else {`

**Review by jhedberg — CHANGES_REQUESTED** [review]
Mostly some style-related comments

**Review by sjanc — COMMENTED** [review]
I have general comment on BTP changes,   please keep in mind that BTP is used by multiple IUTs and should be kept backward compatible,  ie adding random fields to already used BTP commands/events will be difficult

**Comment by CherQin** [comment]
Hi @dleach02, our auto-pts test is based on nxp edgefast host stack, not tested based on zephyr host stack, so will upstream our auto-pts test after verifying it with zephyr host stack in the future.

Hi @jhedberg，@sjanc， thank you very much for your review, will keep your comments in mind and follow it in the future.
For this PR, I'll delete it.
